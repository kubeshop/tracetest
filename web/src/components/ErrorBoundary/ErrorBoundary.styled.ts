import {CloseCircleFilled} from '@ant-design/icons';
import styled from 'styled-components';

export const Container = styled.div`
  height: calc(100vh - 200px);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const Content = styled.div`
  display: flex;
  max-width: 800px;
  padding: 24px;
`;

export const Icon = styled(CloseCircleFilled)`
  color: ${({theme}) => theme.color.error};
  font-size: 32px;
`;
