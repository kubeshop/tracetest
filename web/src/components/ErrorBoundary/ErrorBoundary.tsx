import {Typography} from 'antd';
import * as S from './ErrorBoundary.styled';

interface IErrorBoundaryProps {
  error: Error;
}

const ErrorBoundary: React.FC<IErrorBoundaryProps> = ({error}) => {
  return (
    <S.Container>
      <S.Icon />
      <Typography.Title level={1}>Something went wrong!</Typography.Title>
      <S.Content>{error.toString()}</S.Content>
    </S.Container>
  );
};

export default ErrorBoundary;
