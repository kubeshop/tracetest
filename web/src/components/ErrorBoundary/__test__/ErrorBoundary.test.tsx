import {render} from '@testing-library/react';
import ErrorBoundary from '../ErrorBoundary';

test('ErrorBoundary', () => {
  const result = render(<ErrorBoundary error={new Error('cannot find value of undefined')} />);
  expect(result.container).toMatchSnapshot();
});
