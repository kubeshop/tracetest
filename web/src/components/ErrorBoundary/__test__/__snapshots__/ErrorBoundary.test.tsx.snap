// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary 1`] = `
<div>
  <div
    class="sc-bdvvtL mLXGX"
  >
    <span
      aria-label="close-circle"
      class="anticon anticon-close-circle"
      role="img"
      style="color: red; font-size: 32px;"
    >
      <svg
        aria-hidden="true"
        data-icon="close-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
        />
      </svg>
    </span>
    <h2
      class="ant-typography"
    >
      Something went wrong!
    </h2>
    <div
      style="display: flex; max-width: 800px; padding: 24px;"
    >
      Error: cannot find value of undefined
    </div>
  </div>
</div>
`;
